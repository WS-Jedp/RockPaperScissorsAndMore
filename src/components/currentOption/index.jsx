import React from 'react'
import { IconLizard, IconPaper, IconScissors, IconRock, IconSpock } from '../icons'

export const CurrentOption = ({option = 'rock'}) => {

  const optionName = option.toLowerCase(); 

    switch (optionName) {
      case "rock":
        return <IconRock />
        break;
      case "paper":
        return <IconPaper />
        break;
      case "scissors":
    return <IconScissors />
        break;
      case "lizard":
        return <IconLizard />
        break;
      case "spock":
        return <IconSpock />
        break;
    
      default:
        return <IconRock />
        break;
    }
  }