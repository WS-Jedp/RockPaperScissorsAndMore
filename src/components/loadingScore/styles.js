import styled from 'styled-components'
import { LoadingScoreAnimation } from '../../styles/Animation'

export const LoadingWrapper = styled.div`
  position: relative;
  margin: 9px;
  width: 30px;
  height: 30px;
  background-color: var(--white-color);
  border: 1px solid var(--blue-color);
  border-left: 0; 
  border-bottom: .6px;
  border-radius: 50%;
  ${LoadingScoreAnimation('infinite', '.6s', 'linear')};
`
