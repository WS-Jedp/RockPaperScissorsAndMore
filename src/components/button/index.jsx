import React from 'react'
import { ButtonSecondaryWrapper, ButtonPrimaryWrapper } from './styles'

export const ButtonSecondary = ({title = 'Rules', action = () => {}}) => (
  <ButtonSecondaryWrapper onClick={action}>
    {title}
  </ButtonSecondaryWrapper>
)

export const ButtonPrimary = ({title = 'Play Again', action = () => {}}) => (
  <ButtonPrimaryWrapper onClick={action}>
    {title}
  </ButtonPrimaryWrapper>
)
