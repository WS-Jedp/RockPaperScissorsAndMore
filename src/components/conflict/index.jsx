import React from 'react'
import { ConflictWrapper } from './styles'

import { PickedOption } from '../pickedOption'
import { PlayAgain } from '../playAgain'

export const Conflict = ({userOption = {}, homeOption = {}}) => {

  const isMobile = false
  
  return (
    <>
      <ConflictWrapper>
        <PickedOption title='You Picked' option={userOption.name} color={userOption.color} picked={true} />
        {
        isMobile && <PlayAgain title='You Win'action={() => {}} />
        }
        <PickedOption title='The House Picked' option={homeOption.name} color={homeOption.color} />
      </ConflictWrapper>
      {
        !isMobile && <PlayAgain title='You Win'action={() => {}} />
      }
    </>
  )
}
