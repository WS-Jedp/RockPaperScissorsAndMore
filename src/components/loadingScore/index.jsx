import React from 'react'
import { LoadingWrapper } from './styles'

export const LoadingScore = () => (
  <LoadingWrapper />
)
