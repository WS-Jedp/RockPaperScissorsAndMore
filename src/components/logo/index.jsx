import React from 'react'
import logoSVG from '../../assets/images/logo-bonus.svg'

import { LogoWrapper } from './styles'

export const Logo = () => (
  <LogoWrapper aria-label="Logo Rock, Paper, Scissors And More">
    <img src={logoSVG} alt="Logo Rock, Paper, Scissors And More"/>
  </LogoWrapper>
)