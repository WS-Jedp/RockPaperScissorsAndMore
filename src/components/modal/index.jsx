import React from 'react'
import ReactDOM from 'react-dom'
import { ModalWrapper } from './styles'

export const Modal = ({children}) => ReactDOM.createPortal((
  <ModalWrapper>
    { children || 'Hello World' }
  </ModalWrapper>), 
  window.document.getElementById('modal-portal'))
