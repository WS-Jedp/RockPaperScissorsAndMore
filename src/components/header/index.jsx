import React from 'react'
import { WrapperHeader } from './styles'

import { Score } from '../score/index'
import { Logo } from '../logo'

export const Header = () => (
  <WrapperHeader>
    <Logo />
    <Score />
  </WrapperHeader>
) 