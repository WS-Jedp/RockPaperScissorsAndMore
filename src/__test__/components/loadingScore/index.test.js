import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import { LoadingScore } from '../../../components/loadingScore'

describe('Tests for the "LoadingScore" component', () => {

  const LoadingScoreComponent = create(<LoadingScore />)
  const LoadingScoreMount = mount(<LoadingScore />)

  test('should the component be the same visual', () => {
    expect(LoadingScoreComponent.toJSON()).toMatchSnapshot()
  })

  test('should the length of the component be one', () => {
    expect(LoadingScoreMount.length).toBe(1)
  })

})
