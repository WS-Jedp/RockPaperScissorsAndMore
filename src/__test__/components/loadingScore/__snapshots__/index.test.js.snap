// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the "LoadingScore" component should the component be the same visual 1`] = `
<div
  className="sc-AxjAm eViVcq"
/>
`;
